graphql/src/main/java/com/scalar/db/graphql/instrumentation/TransactionInstrumentation.java

```diff
@@ -78,7 +80,21 @@ public TransactionInstrumentation(DistributedTransactionManager transactionManag
     Argument commitArg = directive.getArgument("commit");
 
     DistributedTransaction transaction;
+
+    String txId;
     if (txIdArg == null) {
+      txId = null;
+    } else {
+      Value<?> value = txIdArg.getValue();
+      if (value instanceof StringValue) {
+        txId = ((StringValue) value).getValue();
+      } else if (value instanceof VariableReference) {
+        txId = (String) executionContext.getVariables().get(((VariableReference) value).getName());
+      } else {
+        throw new IllegalStateException("Unexpected value of txId: " + value);
```

Comment:
Normally, execution will never reach this point since the document is validated for the schema in advance and the directive argument value can be safely cast to `StringValue` or `VariableReference` of a string.